pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/MultipleTreesBuilder.java

```diff
@@ -115,13 +120,33 @@ public MultipleTreesBuilder(@Nullable List<StarTreeIndexConfig> indexConfigs, bo
     }
   }
 
+  private SeparatedStarTreesMetadata getExistingStarTrees()
+      throws Exception {
+    try {
+      if (_metadataProperties.containsKey(MetadataKey.STAR_TREE_COUNT)) {
+        // Extract existing startrees
+        // clean star-tree related files and configs once all the star-trees are separated and extracted
+        SeparatedStarTreesMetadata existingStarTrees = extractStarTrees();
+        StarTreeBuilderUtils.removeStarTrees(_indexDirectory);
+        _metadataProperties.refresh();
+        return existingStarTrees;
+      } else {
+        return new SeparatedStarTreesMetadata(_segmentDirectory);
```

Comment:
Consider returning `null` here to avoid overhead (`null` basically represents there is no existing star-tree)

Reply:
makes sense, done.